### DISCUSSION TITLE
[help] my contributions not shown

### DISCUSSION THREAD

**Discussion by patrickjonesuk** [community/community#22657] [New to GitHub]
<p><a href="https://github.com/Skybot-dev/Skybot-rewrite/graphs/contributors" rel="nofollow noopener">this reposiitory</a> shows that I (pjones123) have only made 2 commits. I have actually made 9 on the current master branch, but also many more on other now deleted branches, which is shown <a href="https://github.com/Skybot-dev/Skybot-rewrite/commits?author=pjones123" rel="nofollow noopener">here</a>. It appears to count commits by other users correctly, including counting commits on now deleted branches.<br>
Can anyone advise how I can get all of my commits to show in the graph above?</p>

**Comment by lee-dohm** [comment]
<p>Hello and welcome to our community!</p>
<p>If you notice at the top of the commit graph that you link to, it states:</p>
<blockquote>
<p>Contributions to master, excluding merge commits</p>
</blockquote>
<p>If you execute the command:</p>
<pre><code class="lang-shell">git log --author=pjones123 --no-merges
</code></pre>
<p>You’ll see the following output:</p>
<pre><code class="lang-nohighlight">commit 294f0fe7523b041319195ffb3c9eb21b19963079
Author: pjones123 &lt;64071912+pjones123@users.noreply.github.com&gt;
Date:   Wed Aug 26 18:51:43 2020 +0100

    removed print statements and timer

commit 52f8fbf00468bb4c609931756eff2960a1a3b14e
Author: pjones123 &lt;64071912+pjones123@users.noreply.github.com&gt;
Date:   Tue Aug 25 17:22:35 2020 +0100

    removed !r as alias
    
    too general for that, and !rem is quick enough.
</code></pre>
<p>So the graph is correct given the two restrictions stated. Unfortunately, because of this, there isn’t a way to show the other commits you’ve submitted in the given graph.</p>
<p>On the other hand, I do see a number of other commits attributed to “Patrick Jones”. If those commits were authored by you also, you could add the email address associated with those commits to your GitHub account and they would then be counted as authored by your <a class="mention" href="/pjones123">@pjones123</a> account. You can check <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-profile/why-are-my-contributions-not-showing-up-on-my-profile">our help documentation on properly configuring your account to attribute commits</a> for more information.</p>
<p>Let us know if you have more questions.</p>

**Comment by patrickjonesuk** [comment]
<p>Thanks for the response! The commits from “Patrick Jones” are indeed from me as well. I presume the difference is due to the fact that those commits were made from my pc using git bash rather than online? How would I go about making those show for the same account? They have the same email, and as far as I’m aware, they are the same account? I was able to make commits with it while the repo was private, so they must be the same? Thanks for any more help you can provide.</p>

**Comment by lee-dohm** [comment, marked as answer]
<aside class="quote no-group" data-username="pjones123" data-post="3" data-topic="129563">
<div class="title">
<div class="quote-controls"></div>
 pjones123:</div>
<blockquote>
<p>They have the same email, and as far as I’m aware, they are the same account?</p>
</blockquote>
</aside>
<p>When those commits were created, you were using a different email address for them. For example:</p>
<p>Commits attributed to <a class="mention" href="/pjones123">@pjones123</a>:</p>
<pre><code class="lang-nohighlight">commit 52f8fbf00468bb4c609931756eff2960a1a3b14e
Author: pjones123 &lt;64071912+pjones123@users.noreply.github.com&gt;
Date:   Tue Aug 25 17:22:35 2020 +0100

    removed !r as alias
    
    too general for that, and !rem is quick enough.
</code></pre>
<p>Commits attributed to “Patrick Jones”:</p>
<pre><code class="lang-nohighlight">commit ef1aa4b7ed391ad8620e0ae2406a979da37fa079
Author: Patrick Jones &lt;REDACTED@REDACTED.REDACTED&gt;
Date:   Tue Aug 25 16:32:15 2020 +0100

    made !price case-insensitive
</code></pre>
<p>You can get the exact email address used by using the <code>git log</code> command in your local copy of the repository. I didn’t include it here for privacy reasons.</p>
<p>So, as I mentioned in my previous reply, <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-profile/why-are-my-contributions-not-showing-up-on-my-profile#you-havent-added-your-local-git-commit-email-to-your-profile">you would need to add the email address associated with those commits to your GitHub account</a>.</p>

**Comment by patrickjonesuk** [comment]
<p>Thank you so much! I added the email address to my GitHub account and it all seems to be working. Still slightly confused as to how I had write access to the repo while it was private, but eh, it works now so that doesn’t really matter. Thanks again!</p>

**Comment by canuckjacq** [comment]
<aside class="quote no-group" data-username="pjones123" data-post="5" data-topic="129563">
<div class="title">
<div class="quote-controls"></div>
 pjones123:</div>
<blockquote>
<p>Still slightly confused as to how I had write access to the repo while it was private</p>
</blockquote>
</aside>
<p>No need to worry. Your commit email address doesn’t form part of your credential access. Your credentials (username and password/personal access token or SSH key) are likely stored in Keychain or similar. The email address you use for your commits is independent of that.</p>
